### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by kiranmaya** [Blazor-Diagrams/Blazor.Diagrams#203]
Hi,
after creating the add two number nodes, how to execute it?.
say it has resulted in, how to read that value.

**Comment by kiranmaya** [comment]
@zHaytam can provide me a simple example of two nodes.I don't know where to start, how to connect two instances in a sequence etc.

**Comment by 314159265meow** [comment]
Hello, as part of the evaluation of this library I did something like that.
![Animation](https://user-images.githubusercontent.com/13624874/175913085-7b641632-07f9-4540-9e17-b39ef725988c.gif)

The problem is that it was part of a bigger project and I can't just post the code here. I will try to give you a gist of how I did it:

Each node is a custom node that implements a `Process()` method.  
The ports are custom ports with an `IsInput` and a `Value` property. 

"Run Engine" selects all starting nodes (those with 0 links on the input ports and at least one output link). and calls the `Process()` method for each of these. In the process method the node either uses it's state (e.g. value of the checkbox or input field) or the `Value` property of it's connected ports to determine it's own `Value`.  
After that I take all connected nodes from the current nodes, do the same process again till there are no nodes left.

This is not the best way to do it, usually you want your data separate from your presentation, e.g. to run your "engine" without graphical representation, but it was fun and quick to implement. I will see if I can copy the relevant parts to a new project and post it here later.

**Comment by kiranmaya** [comment]
@314159265meow 
Yep , this is exactly what i am looking for .plz share if feasible.
`After that, I take all connected nodes from the current nodes, and do the same process again till there are no nodes left.
`
I m not getting this , any ideas, on how to move value from one node to another node. say I have List<Action> nodesProcess, here looping and process calling is fine, but how to out pin value to, which output pin from a current node is transferred to next node input.

**Comment by kiranmaya** [comment]
@314159265meow Thanks for the share, will check it out.
